src/mlpack/tests/main_tests/hmm_training_tests.cpp

```diff
@@ -0,0 +1,209 @@
+/**
+ * @file hmm_training_tests.cpp
+ * @author Daivik Nema
+ *
+ * Test mlpackMain() of hmm_train_main.cpp.
+ */
+#include <string>
+#include <fstream>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "HMMTrain";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/hmm/hmm_train_main.cpp>
+#include <mlpack/methods/hmm/hmm_model.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+using namespace mlpack;
+
+struct HMMTrainMainTestFixture
+{
+ public:
+  HMMTrainMainTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~HMMTrainMainTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(HMMTrainMainTest, HMMTrainMainTestFixture);
+
+inline void fileExists(std::string fileName)
```

Comment:
I know the style checker is not running right now, but I hope to have that back online in the next day or two.  It can tell you about these things automatically.  In any case, can we call this function `FileExists()` so it matches the style guide?